File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXAnr_NdkDisabledScenario.java
package com.bugsnag.android.mazerunner.scenarios;

import com.bugsnag.android.Configuration;

import android.content.Context;

import androidx.annotation.NonNull;

public class CXXAnr_NdkDisabledScenario extends CXXAnrScenario {

    public CXXAnr_NdkDisabledScenario(@NonNull Configuration config, @NonNull Context context) {
        super(config, context);
        config.getEnabledErrorTypes().setAnrs(true);
        config.getEnabledErrorTypes().setNdkCrashes(false);
    }
}

File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXAnrNdkDisabledScenario.kt
package com.bugsnag.android.mazerunner.scenarios

import android.content.Context
import com.bugsnag.android.Configuration

/**
 * Stops the app from responding for a time period
 */
internal class CXXAnrNdkDisabledScenario(
    config: Configuration,
    context: Context,
    eventMetadata: String?
) : Scenario(config, context, eventMetadata) {

    companion object {
        init {
            System.loadLibrary("cxx-scenarios")
        }
    }

    init {
        config.enabledErrorTypes.anrs = true
        config.enabledErrorTypes.ndkCrashes = false
    }

    external fun crash()

    override fun startScenario() {
        super.startScenario()
        crash()
    }
}
